java/ql/src/experimental/Security/CWE/CWE-1004/SensitiveCookieNotHttpOnly.ql

```diff
@@ -93,14 +88,24 @@ predicate setHttpOnlyInCookie(MethodAccess ma) {
       isBooleanTrue(mpa.getArgument(i))
     )
   )
-  or
-  exists(MethodAccess mca |
-    ma.getMethod().calls(mca.getMethod()) and
-    setHttpOnlyInCookie(mca)
-  )
 }
 
-/** Holds if the method or a wrapper method removes a cookie. */
+/**
+ * A taint configuration tracking flow of a method or a wrapper method that sets
+ * the `HttpOnly` flag.
+ */
+class SetHttpOnlyInCookieConfiguration extends TaintTracking2::Configuration {
+  SetHttpOnlyInCookieConfiguration() { this = "SetHttpOnlyInCookieConfiguration" }
+
+  override predicate isSource(DataFlow::Node source) { any() }
```

Comment:
For performance reasons you should always constrain sources and sinks of taint flow configurations, because I think all taint flows are calculated and then you get to ask if there was a flow from A to B. You should restrict the source to be the qualifier of `ma` which satisfies `setHttpOnlyInCookie` or `removeCookie`. Then you don't need to define `ma2` later where this is used.

Reply:
Thanks @owen-mc. I've made the change and the code is much more cleaner.